client.go

```diff
@@ -468,3 +453,20 @@ func (c *Client) start() {
 		}
 	}
 }
+
+// SetEventsHeaderCallback takes a response header name and callback function
+// on SendDataToMoesif, the callback headers are read from the response and
+// passed to the callback functions
+func (c *Client) SetEventsHeaderCallback(header string, callback func(string)) {
+	c.eventHeaderCallbacks[header] = callback
+}
+
+// notify iterates over event header callbacks, looks up the header value, and calls back
+// if a non empty value is found
+func (c *Client) notify(headers http.Header) {
```

Comment:
eTag updates are pushed out from all responses here and processed by the receiver if necessary